loopy/check.py

```diff
@@ -307,10 +310,12 @@ def check_for_data_dependent_parallel_bounds(kernel):
         parameters = set(dom.get_var_names(dim_type.param))
         for par in parameters:
             if par in kernel.temporary_variables:
-                raise LoopyError("Domain number %d has a data-dependent "
-                        "parameter '%s' and contains parallel "
-                        "inames '%s'. This is not allowed (for now)."
-                        % (i, par, ", ".join(par_inames)))
+                warn_with_kernel(kernel, "data_dep_par_bound('%s')" % par,
```

Comment:
PyOP2 and Firedrake tests are passing even though we get this error, so I changed the error to a warning. This might need a more thorough investigation, but that was out of my depth for the moment.